### ISSUE TITLE
"None of your items need to be repaired"

### ISSUE THREAD

**Issue by pjmagee** [BosslandGmbH/BuddyWing#42]
Buddywing is constantly trying to repair items. The repair all button is greyed out, so it cannot be clicked. All it tries to do is go to the medical droid and repair.

**Comment by aevitas** [comment]
Sounds like the expression used to determine repairs isn't evaluating properly. We've had this in the past when one or more items didn't have durability - I'll have a look into it.

**Comment by pjmagee** [comment]
I'm not sure what is currently being done, but a solution might be to check the state of the "Repair all" button, if its greyed out, then don't do anything and then you are offloading more logic to the client?

Just an idea.

**Comment by CryoGenesisX** [comment]
Hmm, if the bot thinks it needs repairs, but my gear is 100%, I don't think you want that check at the vendor. You want that to happen in every x cycles. Also I don't know what the current min value is for repair. I'd say yellow armor?

**Comment by pjmagee** [comment]
Yeah I like that idea, checking to see if the Yellow Armor "buff" icon is on the player.

**Comment by pjmagee** [comment]
However, maybe as a fall back option as a fail safe, there could be a "verification" check. If the bot does go to the vendor and it IS greyed out after attempting a repair. It should log this so that an investigation can be made as to why the first logic to determine if a repair was needed has now started to fail?
